backend/src/main/java/trn/logistics/knapsack/exporter/PdfExporter.java

```diff
@@ -0,0 +1,54 @@
+package trn.logistics.knapsack.exporter;
+
+import be.quodlibet.boxable.BaseTable;
+import be.quodlibet.boxable.datatable.DataTable;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.common.PDRectangle;
+import org.springframework.stereotype.Component;
+import trn.logistics.knapsack.dto.KnapsackSolution;
+import trn.logistics.knapsack.dto.Vehicle;
+
+import java.io.IOException;
+
+@Component
+@Slf4j
+public class PdfExporter {
+
+    public PDDocument createPdfFromKnapsack(KnapsackSolution kns) throws IOException {
+
+        PDDocument mainDocument = new PDDocument();
+        for (Vehicle vehicle : kns.getSelectedVehicles()) {
+            PDPage myPage = new PDPage(PDRectangle.A4);
+            //TODO add VEHICLE details to PDF
+            //TODO Change Table to table with header
+
+            BaseTable dataTable = createBaseTable(myPage, mainDocument);
+
+            DataTable myTable = new DataTable(dataTable, myPage);
+                myTable.addListToTable(vehicle.getContainedElementsListForPDF(), DataTable.NOHEADER);
+                dataTable.draw();
+
+            mainDocument.addPage(myPage);
+
+            mainDocument.save(kns.getName() + ".pdf");
```

Comment:
Don't save to filesystem, for testing add another method.
Also this save is within the for-loop! Though if the kns contains 10 vehicles we would save 10x